project-minimization-plugin/src/main/kotlin/org/plan/research/minimization/plugin/services/MinimizationPsiManagerService.kt

```diff
@@ -139,10 +133,15 @@ class MinimizationPsiManagerService {
                 KotlinElementLookup.lookupExpected(element).process(context, psiCache)
                     .map { PsiIJEdge.UsageInPSIElement(from, it) }
             }
-            InstanceLevelGraph(
-                vertices = (nodes.map(Pair<*, PsiStubDDItem>::second) + fileHierarchyNodes).distinct(),
-                edges = psiEdges + overloadEdges + usageEdges + obligatoryOverride + fileHierarchyEdges + fileEdges + expectActual,
-            )
+
+            val vertices = (nodes.map(Pair<*, PsiStubDDItem>::second) + fileHierarchyNodes).distinct()
+            val edges = psiEdges + overloadEdges + usageEdges + obligatoryOverride + fileHierarchyEdges + fileEdges + expectActual
+            val builder = DirectedPseudograph.createBuilder<PsiStubDDItem, PsiIJEdge>(PsiIJEdge::class.java)
+
+            vertices.forEach { builder.addVertex(it) }
+            edges.forEach { builder.addEdge(it.from, it.to, it) }
```

Comment:
Do we then actually need the information `from -> to` in the edge in that case? I'm not fluent with the library I've used, but maybe it provides a way to get all adjacent nodes with an edge information?

Reply:
To be honest -- no, but I've tried to leave everything without big changes. I will optimize/refactor the idea part of the algorithm a bit later